core/block.ts

```diff
@@ -319,6 +319,20 @@ export class Block implements IASTNodeLocation {
     }
   }
 
+  /**
+   * Returns true if this block is the same block as the given block or contains
+   * the provided block as a descendent.
```

Comment:
Can you comment here that this includes all next blocks? People are easily confused about descendants vs children.

Reply:
Actually, that reminds me: the intent of this PR is to detect whether the selected block would get deleted if another block is deleted, but the next blocks of the originally deleted block do NOT get deleted (and the associated stack gets "healed"). I want to include the next blocks of inputs, but not the next blocks in the same stack as the deleted block. So this implementation probably isn't exactly right?

Reply:
Hmm my instinct is that it's not actually possible to run into that case since this is specifically a problem for shadow blocks being selected while highlighting another block, and you can't have non-shadows attached to shadows, so no healing would occur.

Basically the healing case is incompatible with the shadow selection case, iiuc. But it's a bit complex so I'm not 100%

Reply:
On the other hand, if the deleted block comes between the selected shadow block and its highlighted ancestor, then deleting the block could sever the connection even if the selected block itself did not get deleted in the process. In which case, technically I think we should remove the highlight, delete the block and its inputs, then re-highlight an appropriate ancestor of the selected shadow block based on the new stack arrangement?

Reply:
Is it possible for the deleted block to come between the selected shadow block and its ancestor and the selected shadow block not get deleted?

Reply:
Yeah, maybe I'm overthinking it.

However, I decided to remove the new `contains` function that I added, and manually inlined it into the one place I was using it. I figured it wasn't worth adding a new public function to the API that may be confusing and would need to be maintained indefinitely.